### DISCUSSION TITLE
Extracting the "supply chains" from an OpenTTD saved game file

### DISCUSSION THREAD

**Discussion by michalc** [OpenTTD/OpenTTD#11462] [Q&A]
I'm trying to extract some sort of view of the "supply chains" from the OpenTTD saved game file. So some sort of graph/network of "what industries/towns supply what other industries/towns"

I can use the (really good) https://github.com/TrueBrain/OpenTTD-savegame-reader project to get extract the chunks...

... but then I am slightly overwhelmed/not really sure where to begin in terms of which chunks to look in/what they all mean. Does anyone have any pointers?

Thanks,

Michal

**Comment by PeterN** [comment]
Savegames do not store this information. Definitions for cargo types themselves, and what house, industries and industrytiles accept / produce all come from either data inside OpenTTD or from NewGRF files.

**Reply by michalc** [reply]
Ah so I don't think I'm looking for "in general, what do industries produce/accept", but more "what is currently being sent where, in this particular game"

**Reply by PeterN** [reply]
Then you need to look at vehicles and their orders. This will only tell you what is currently being moved at a point in time, and will only tell you stations. The information that links stations to industries and towns is built at runtime based on tile location, etc.

The link graph is used to guide this when cargo distribution is enabled, but doesn't directly correlate.

**Reply by michalc** [reply]
> The information that links stations to industries and towns is built at runtime based on tile location, etc.

Ah understood - thanks

I think maybe for my purposes looking at the link graph _might_ be enough... it's a sort of a rough "proxy" for industries.

**Comment by michalc** [comment]
So as soon as I posted that, I found https://github.com/TrueBrain/OpenTTD-savegame-reader/blob/main/linkgraph/__main__.py, so it looks like `LGRP` might be a good chunk to investigate first (and even use the code in that file)

<img width="61" alt="Screenshot 2023-11-16 at 09 50 17" src="https://github.com/OpenTTD/OpenTTD/assets/13877/9bafc43d-992a-4631-b3c3-778df1cbb6d9" alt="Rubber duck">
